client/src/components/generic/CheckPermissions.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Grid from '@material-ui/core/Grid';
+import { Box, Typography } from '@material-ui/core';
+import { checkPermissions } from '../../utils';
+
+export const CheckPermissions = ({ roles, permittedRoles, children, renderMessage = false }) => {
+  // If user has a permitted role assigned, grant access to components
+  if (checkPermissions(roles, permittedRoles)) {
+    return children;
+  }
+
+  // Optional: display message if no access granted by role(s)
+  if (renderMessage) {
```

Comment:
Maybe `renderErrorMessage` would be more appropriate here?

Reply:
Coming next commit! :)